src/Explorer/Panes/SaveQueryPanel/index.tsx

```diff
@@ -0,0 +1,174 @@
+import { useBoolean } from "@uifabric/react-hooks";
+import { Text, TextField } from "office-ui-fabric-react";
+import React, { FunctionComponent, useState } from "react";
+import { Areas, SavedQueries } from "../../../Common/Constants";
+import { getErrorMessage, getErrorStack } from "../../../Common/ErrorHandlingUtils";
+import { Query } from "../../../Contracts/DataModels";
+import { Action } from "../../../Shared/Telemetry/TelemetryConstants";
+import { traceFailure, traceStart, traceSuccess } from "../../../Shared/Telemetry/TelemetryProcessor";
+import { logConsoleError } from "../../../Utils/NotificationConsoleUtils";
+import Explorer from "../../Explorer";
+import QueryTab from "../../Tabs/QueryTab";
+import { GenericRightPaneComponent, GenericRightPaneProps } from "../GenericRightPaneComponent";
+
+interface SaveQueryPanelProps {
+  explorer: Explorer;
+  closePanel: () => void;
+}
+
+export const SaveQueryPanel: FunctionComponent<SaveQueryPanelProps> = ({
```

Comment:
Do we have a unit test for this? At least an enzyme snapshot check?

Reply:
Added testcases and enzyme snapshot